fastmri/data/mri_data.py

```diff
@@ -260,16 +265,24 @@ def __init__(
         else:
             dataset_cache = {}
 
+        # obtain annotation
+        if annotations is True:
+            annotation_fileName = self.Download_csv(annotations_version)
+
         # check if our dataset is in the cache
         # if there, use that metadata, if not, then regenerate the metadata
         if dataset_cache.get(root) is None or not use_dataset_cache:
             files = list(Path(root).iterdir())
             for fname in sorted(files):
                 metadata, num_slices = self._retrieve_metadata(fname)
 
-                self.examples += [
-                    (fname, slice_ind, metadata) for slice_ind in range(num_slices)
-                ]
+                csv_data = pd.read_csv(annotation_fileName)
+                annotations = csv_data[csv_data['file'] == fname.stem]
+
+                if len(annotations) > 0:
+                    self.examples += [
+                        (fname, slice_ind, metadata) for slice_ind in range(num_slices)
+                    ]
```

Comment:
You can probably add the annotations to the `metadata` field. You can use a small test case to see how it is. I think it is a `dict`.

Reply:
Hi Matt, I have three questions for adding the annotations to the metadata:

1. To confirm, does the slice_ind in `self.examples` correspond to the second column "slice" in the csv?
2. If the same slice has more than one annotations, should we add all of them in metadata and how would you suggest to organize them?
e.g. 
file1000001,20,No,197,188,34,8,Cartilage - Partial Thickness loss/defect
file1000001,20,No,264,122,17,39,Ligament - MCL Low-Mod Grade Sprain
3. Should I return all columns(file,slice,study_level,x,y,width,height,label) in csv as tuple into metadata?

Reply:
Hello @Gaskell-1206 - hope you had a good Thanksgiving :).

1. Yes I am pretty sure they are the same. I prefer `slice_ind` because `slice` is a Python core library function.
2. Yes you should add all of them to the metadata. One thing to keep in mind with metadata is we need to make sure PyTorch can collate all of this stuff for batching. This means using simple data structures and types. I would just have a `dict` in metadata like this:
```python
metadata['annotation'] = {
    'fname': 'file1000001',
    'slice': 20,
    'study_level': 'No',
    'x': 264,
    'y': 188,
    'width': 34,
    'height': 8,
    'label': 'Cartilage - Partial Thickness loss/defect'
}
```
3. Use the `dict` structure - the default PyTorch `collate_fn` has a tree-traversal functionality built-in to go over the dict, convert everything to Tensors, and batch them. By using simple string and integer types I think this will be robust and stable.

Reply:
Hi Matt,

Hope you had a good Thanksgiving, too!

For practical use,  I'm wondering how to deal with annotation that have the  same `fliename` and `slice`.
e.g. 
file1000638,21,No,235,176,13,6,Meniscus Tear
file1000638,21,No,63,112,18,50,Ligament - MCL Low-Mod Grade Sprain

In my code, I use `filename` and `slice` from `self.examples` as index to obtain annotations from csv.

When adding them into dict, I have no idea which way is better for dataloader:
1. add them as `list` inside `dict`:
metadata['annotation'] = {'fname': 'file1000638',
                                              'slice': 21,
                                              'study_level': 'NO',
                                              'x': [235,63],
                                              'y': [176,112],
                                              'width': [13,18],
                                              'height': [6,50],
                                              'label': ['Meniscus Tear', 'Ligament - MCL Low-Mod Grade Sprain']}
2.add several `dict` as `tuple`:
metadata['annotation'] = tuple(dict1, dict2, ...)

I will submit a draft pull for I have done now. 
Thanks!

Reply:
Hello @Gaskell-1206, this presents a problem for batching. You can add a new parameter, e.g., `multiple_annotation_policy`. We could include three options: `'first'`, `'random'` or `'all'`. If `'first'`, then only use the first annotation. If `'random'`, then pick an annotation at random. If `'all'`, then extend `self.examples` to have two copies of the same slice, one for each annotation.

The issue is if you have something that is variable-length then you may not be able to batch them together.

Also another note on batching: Make sure you put in dummy values for all fields if there is no annotation for that slice.

Reply:
I see your points! It's a good way for batching.

Thanks for your help.

Reply:
Hi Matt, can you tell me how to deal with two situation related to no annotation found?

1. If there are no any annotations for the whole filename, we should remove that file from `self.examples` as you suggested in Slack before.
2. If there is no annotation for some slices in the same filename, we should put in dummy values for all fields. 
For using dummy values, should metadata['annotation'] be like:
{'fname': 'file1000638',
'slice': 1,
'study_level': None,
'x': 0,
'y': 0,
'width': 0,
'height': 0,
'label': None}

Thanks for answering my questions！

Reply:
For 1:

I would just use dummy values for these ones. The user can filter `examples` if they don't want to use these.

For 2:

For dummy values we should use -1 since they should all be non-negative. This will be easy for users to detect.

Don't use the Python `None`. Use an empty string or `"None"`.